src/sinks/greptimedb/service.rs

```diff
@@ -119,10 +126,33 @@ impl GreptimeDBService {
             }))
         };
 
-        // TODO: tls configuration
-
-        GreptimeDBService {
+        Ok(GreptimeDBService {
             client: Arc::new(client),
+        })
+    }
+
+    fn try_from_tls_config(tls_config: &TlsConfig) -> Option<ClientTlsOption> {
+        if let Some(ca_path) = tls_config.ca_file.as_ref() {
+            let cert_path = tls_config.crt_file.as_ref().expect(
```

Comment:
This shouldn't be a panic. Can you change this to return a `Result::Err`? You can use `Snafu` similar to [here](https://github.com/vectordotdev/vector/blob/master/src/nats.rs#L8-L16) to define the errors.

Reply:
hmmm.. I checked other sinks for `expect` usage. But no problem, let me draft an update for this.

Reply:
Ah, if that's the case, we need to update those sinks too.

Invalid settings should result in an error message rather than a panic. This is primarily to handle the situation where a config is reloaded. If you attempt to reload an invalid config it should continue to run with the old config. A panic would kill the whole process.

Reply:
That's reasonable. I had a quick grep and found https://github.com/vectordotdev/vector/blob/master/src/sinks/datadog/metrics/service.rs#L162 that you might want to check.